### DISCUSSION TITLE
Number provisioning modal

### DISCUSSION THREAD

**Discussion by lgalloherran** [twilio-labs/paste#1216] [Office hours] [answered]
I'm working on the number provisioning modals (After you search for the number and press 'buy') and I have some questions regarding [the mockups ](https://www.figma.com/file/hLWQ2TeMdrIrZsnozKNzOa/Buy-a-Phone-Number-PHONE-NUMBERS-Buying-a-number-flow-copy-2?node-id=32%3A21922)I designed: 
[**1. Managing a full-screen modal:** ](https://www.figma.com/proto/hLWQ2TeMdrIrZsnozKNzOa/Buy-a-Phone-Number-PHONE-NUMBERS-Buying-a-number-flow-copy-2?node-id=32%3A21929&viewport=-91%2C456%2C0.20600582659244537&scaling=min-zoom)Should the modal change height depending the content, or should it stay the same?
Can I use an alert in the Lower part of the modal?
[**2. Anchors:**](https://www.figma.com/proto/hLWQ2TeMdrIrZsnozKNzOa/Buy-a-Phone-Number-PHONE-NUMBERS-Buying-a-number-flow-copy-2?node-id=37%3A21338&viewport=-91%2C456%2C0.20600582659244537&scaling=min-zoom) It is ok for those to be near each other? 
[**3. Address drop-down**](https://www.figma.com/proto/hLWQ2TeMdrIrZsnozKNzOa/Buy-a-Phone-Number-PHONE-NUMBERS-Buying-a-number-flow-copy-2?node-id=32%3A21929&viewport=-91%2C456%2C0.20600582659244537&scaling=min-zoom) with detail to provide users more information about address: Can I use this format on a dropdown? Is it accesible?

**Comment by corinne-nagel** [comment, marked as answer]
Hey @lgalloherran , after looking at this design/flow more closely, I don't know if a modal is appropriate for the amount of content you're showing. This is a stepped flow which I think could be solved with a stepper flow/full page design maybe with breadcrumbs at the top. Generally speaking, if an interaction is complex enough to require multiple steps, then it’s complex enough to warrant its own page, and to be afforded the extra space and flexibility that a page provides over an overlay.

1. If you've tested this with users though and the layout works, then having the modal height change based on content is 👍
2. The anchors look fine from what I've seen, which ones specifically are you curious about?
3. While this is technically possible, the combobox content is getting a little long. Similar to tooltips, long strings are difficult for screen readers to understand. It will hear everything in it, but there aren't the same controls as if it were just plain text on the screen where they can go letter by letter, word by word, on long strings if they find it difficult to understand. If possible, can you reduce the amount of copy or put it in a different layout?

**Comment by lgalloherran** [comment]
Hey @corinne-nagel Corinne, Thanks for the quick response. 
I agree it may be better to take user to another page due to the amount of information, but. we have to consider that is a sort of 'portable' flow that appear in different products where taking user away from the actual product is not a good idea. 

1. In the testing I have done with the actual flow, users don't mind to perform the task in a modal and they actually get lost when they are led to another page that is away from the provisioning flow (e.g. Creating a bundle). 
3. I will check another way of selecting this address because users manifested that having only the friendly name will not be clear enough when selecting the address, that's why I proposed to include more information about this address. If you can think of another resource regarding this, it will be great cause I don't have any other idea for now. 
4. I'll check the capabilities display because it looks clickable, but it has been like that since forever. I'll check another option for that. 
5. Also, I wanted to ask you about the alert. Can it be in the lower part of the modal? I put it there because this information is important when confirming the 'buy' with pressing the button.

**Reply by corinne-nagel** [reply]
Got it! Glad you've tested it then, that sounds good 🙂

I think having the alert at the top instead of the bottom makes sense since that is where you've put that pricing info. If you want the alert to be near the button, maybe move that pricing info (the $1/mo) down with the alert so they're together? Did any users have a preference? Might be something to keep an eye out for.

I'll keep thinking on the combobox content, since you're signed up for this week, it'l be good to bring to office hours to chat about and see if the rest of the team has any ideas.

**Reply by corinne-nagel** [reply]
Chatted with the team about some combobox things to think about:

- Could you remove the SID (BU...) number after the address? That seems to make things more lightweight. 
- If you have all of this content in the dropdown, what does it look like when an option is selected? Does it truncate?
- How will the search/auto complete work in the combobox if there is a lot of content in there?
